src/test/java/fi/okm/jod/yksilo/controller/profiili/KiinnostusControllerIT.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2025 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.controller.profiili;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import fi.okm.jod.yksilo.IntegrationTest;
+import fi.okm.jod.yksilo.entity.Yksilo;
+import fi.okm.jod.yksilo.repository.YksiloRepository;
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+/** Integrationtests for {@link KiinnostusController} */
+@AutoConfigureMockMvc
+public class KiinnostusControllerIT extends IntegrationTest {
+
+  @Autowired private YksiloRepository yksilot;
+  @Autowired private MockMvc mockMvc;
+
+  @WithUserDetails("test")
+  @Sql(
+      scripts = {"/data/ammatti.sql"},
+      executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+  @Sql(
+      scripts = {"/data/cleanup.sql"},
+      executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+  @Execution(ExecutionMode.SAME_THREAD)
+  @Transactional
+  @Test
+  void addAmmattiKiinnostus200() throws Exception {
```

Comment:
We have used naming like "shouldOutcomeWhenSometing", e.g. shoudUpdateMuokattuWhenAddingKiinnostukset